### PR TITLE
IN-12119 Make sure the arity is being considered to build the uniqueness key

### PR DESCRIPTION
**Pull request by caioeps**
**What type of change is this?**

- Improvement

**Why is this change necessary?**

- The gem completely disregards the arity when building the uniqueness key. If a job receives 5 arguments and its unique arity is 3, it would still build the unique queue based on the 5 arguments.

**How does it address the issue?**

- By slicing the arguments before 

**What side effects does this change have?**

- None, ideally.

**GIFs and/or Images (in case there are changes in the Admin app)**

- 

**Links (Jira, Confluence, zeplin, etc)?**

- https://intricately.atlassian.net/browse/IN-12056

### CODE REVIEW

File: spec/resque/plugins/unique_by_arity_spec.rb [hunk 1]
```diff
@@ -29,6 +29,36 @@ def self.perform(_req, _opts = {})
...
+          )
+        end
+      end
+
+      it 'generates the same hash regardless of extra arguments' do
+        args_1 = [1, 2, {}]
+        args_2 = [1, 3, {}]
+
+        # require 'byebug'; byebug
+
```
Comment (silviogreuel) [inline spec/resque/plugins/unique_by_arity_spec.rb]: Is this commentary needed?
```suggestion
```

File: spec/fixtures/unique_slow_fake_job_with_arity.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class UniqueSlowFakeJobWithArity
   @queue = :normal
 
-  def self.perform(_req, _opts = {})
-    sleep 0.1
```
Comment (silviogreuel) [inline spec/fixtures/unique_slow_fake_job_with_arity.rb:5]: This is how you make comedy

File: spec/resque/plugins/unique_by_arity_spec.rb [hunk 3]
```diff
@@ -29,6 +29,36 @@ def self.perform(_req, _opts = {})
...
+          def self.perform(_a, _b, _opts = {})
+            # Does something
+          end
+
+          include Resque::Plugins::UniqueByArity.new(
+            arity_for_uniqueness: 1,
+            unique_at_runtime: true,
+            unique_in_queue: true
+          )
```
Comment (silviogreuel) [inline spec/resque/plugins/unique_by_arity_spec.rb]: Can you add tests for:
- Not having `arity_for_uniqueness`
- Having `arity_for_uniqueness` with 2

File: spec/resque_spec.rb [hunk 4]
```diff
@@ -24,6 +24,18 @@
     expect(Resque.size(queue)).to eq 2
   end
 
+  it 'does not enqueue duplicate arguments after the arity limit' do
+    Resque.enqueue UniqueFakeJobWithArity, 'foo', 'bar'
+    Resque.enqueue UniqueFakeJobWithArity, 'foo', 'foo'
+
+    expect(Resque.enqueued?(UniqueFakeJobWithArity, 'foo', 'bar')).to eq true
+    expect(Resque.enqueued?(UniqueFakeJobWithArity, 'foo', 'foo')).to eq true
```
Comment (silviogreuel) [inline spec/resque_spec.rb:32]: Any other resque method that should address a single message by uniqueness?

File: lib/resque/unique_by_arity/modulizer.rb [hunk 5]
```diff
@@ -44,6 +44,12 @@ def self.to_mod(configuration)
               Resque::UniqueByArity.debug("#{self}.unique_in_queue_redis_key for #{args_for_uniqueness} is: #{ColorizedString[key].green}")
               key
             end
+
+            define_method(:unique_in_queue_redis_value) do |queue, payload|
+              unique_hash, args_for_uniqueness = redis_unique_hash(payload)
+              Resque::UniqueByArity.debug("#{self}.unique_in_queue_redis_value for #{args_for_uniqueness} is: #{ColorizedString[unique_hash].green}")
+              unique_hash
+            end
```
Comment (silviogreuel) [inline lib/resque/unique_by_arity/modulizer.rb:52]: Just curious, you are overriding it here? where this method is defined?

**Review by silviogreuel — APPROVED** [review]

File: lib/resque/unique_by_arity/modulizer.rb (hunk 5 above)
Comment (caioeps) [reply lib/resque/unique_by_arity/modulizer.rb:52]: It's defined [here](https://github.com/teamintricately/resque-unique_in_queue/blob/master/lib/resque/plugins/unique_in_queue.rb#L27).

Unfortunately, these gems' architectures are not the best.

File: spec/fixtures/unique_slow_fake_job_with_arity.rb (hunk 2 above)
Comment (caioeps) [reply spec/fixtures/unique_slow_fake_job_with_arity.rb:5]: I was the one who wrote it. 🥲

File: spec/resque/plugins/unique_by_arity_spec.rb (hunk 1 above)
Comment (caioeps) [reply spec/resque/plugins/unique_by_arity_spec.rb]: It is not. Thank you for flagging it.

**Pull request merged**